src/utils/ogcServices.ts

```diff
@@ -0,0 +1,206 @@
+import type { Resource } from '@/model/resource'
+import { detectOgcService } from '@datagouv/components-next'
+
+export type OGC_SERVICE_FORMAT = 'wms' | 'wfs'
+
+/**
+ * Information about an OGC service layer (WFS/WMS)
+ */
+export interface OgcLayerInfo {
+  url: string
+  format: OGC_SERVICE_FORMAT
+  layerName?: string
+  title?: string
+}
+
+export interface OgcSearchResult {
+  layerInfo: OgcLayerInfo | null
+  foundWfs: boolean
+}
+
+/**
+ * Helper to parse XML string into a DOM Document
+ */
+export function parseXml(xmlString: string): Document {
+  const parser = new DOMParser()
+  const doc = parser.parseFromString(xmlString, 'text/xml')
+
+  // Check for parsing errors
+  const parserError = doc.querySelector('parsererror')
+  if (parserError) {
+    throw new Error(`XML parsing failed: ${parserError.textContent}`)
+  }
+
+  return doc
+}
+
+/**
+ * Extracts base URL by removing GetCapabilities query parameters
```

Comment:
`service` is not limited to `request=GetCapabilities`, so "GetCapabilities parameters" is misleading (used here and in some tests). I'm surprised it works without the `service` param actually (should be "as required" as `version`).